### PR TITLE
boto3

### PR DESCRIPTION
**Pull request by ltkum**
Table `shorturl` has about 38 million entries, growing with 25'000 entries a day and table `geoadmin-file-storage` has 3.8 million entries.

Link: https://mf-chsdi3.int.bgdi.ch/ltkum_boto_2_to_3

[Demo in map.geo.admin.ch using this branch](http://map.geo.admin.ch/?api_url=//mf-chsdi3.int.bgdi.ch/ltkum_boto_2_to_3)

See https://github.com/geoadmin/mf-chsdi3/issues/3284

KML (S3 + DynamoDB)
* [x]  Create (POST /files)
* [x]  Update (POST /files/{id})
* [x]  Delete (DELETE /files/{id})
* [x]  Read (GET /files/{id}) [ununsed ]

Shortener (DynamoDB)
* [x] Shorten (/shorten.json)
* [x] Redirect (/shorten/{id})

Demo 

[shorten](https://mf-chsdi3.int.bgdi.ch/ltkum_boto_2_to_3/shorten.json?url=https:%2F%2Fmf-geoadmin3.int.bgdi.ch%2F%3FX%3D164565.22%26Y%3D620538.74%26zoom%3D2%26lang%3Den%26topic%3Dlubis%26bgLayer%3Dch.swisstopo.pixelkarte-grau%26catalogNodes%3D1179,1180,1184,1186%26layers%3Dch.swisstopo.lubis-bildstreifen)
[redirect](https://mf-chsdi3.int.bgdi.ch/ltkum_boto_2_to_3/shorten/mvt)

### CODE REVIEW

**Comment by procrastinatio** [comment]
@ltkum Just for reference #3215

**Comment by procrastinatio** [comment]
Deployed and tested. Looks good

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+boto3
```
Comment (procrastinatio) [inline requirements.txt]: Pin the version, aslo botocore

**Comment by procrastinatio** [comment]
Locally, it also works with `Python3`

File: chsdi/views/admin.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 from pyramid.view import view_config
 from pyramid.renderers import render_to_response
 from chsdi.models.clientdata_dynamodb import get_dynamodb_table
+from boto3.dynamodb.conditions import Key
 
 LIMIT = 50
```
[the remark this answers is not in the document]
Comment (ltkum) [reply chsdi/views/admin.py]: In boto3, the dynamodb table.query method gives us the parameter KeyConditionExpression as a non deprecated parameter to apply conditions to keys for the query. Those conditions are in the form of Key('Name').method('what_we_compare_to'). Keys, in this context, are purely a filtering parameter. 
(For what I understood, keys in boto2 and boto3 are different)

**Comment by boecklic** [comment]
@ltkum can you pls rebase this PR on `master` and create a PR with 1 squashed commit? Thnx!

**Comment by procrastinatio** [comment]
Note, test ' lausanne                  vd' != 'lausanne vd' fails everywhere.

**Comment by ltclm** [comment]
> Note, test ' lausanne vd' != 'lausanne vd' fails everywhere.
has been fixed was related to sphinx issue, has been solved

**Pull request merged**